File: brute_force_attack.java
import java.security.*;

public class BruteForceAttack {
    public static String hashPassword(String password) throws NoSuchAlgorithmException {
        MessageDigest md = MessageDigest.getInstance("MD5");
        byte[] hashBytes = md.digest(password.getBytes());
        StringBuilder sb = new StringBuilder();
        for (byte b : hashBytes) sb.append(String.format("%02x", b));
        return sb.toString();
    }

    public static void bruteForce(String targetHash, String charset, int maxLength) throws Exception {
        for (int length = 1; length <= maxLength; length++) {
            generateCombinations("", charset, length, targetHash);
        }
    }

    private static void generateCombinations(String prefix, String charset, int length, String targetHash) throws Exception {
        if (length == 0) {
            if (hashPassword(prefix).equals(targetHash)) {
                System.out.println("[+] Password found: " + prefix);
                System.exit(0);
            }
            return;
        }
        for (int i = 0; i < charset.length(); i++) {
            generateCombinations(prefix + charset.charAt(i), charset, length - 1, targetHash);
        }
    }

    public static void main(String[] args) throws Exception {
        bruteForce("900150983cd24fb0d6963f7d28e17f72", "abc", 3);
    }
}

File: brute_force_attack.py
import itertools
import hashlib

def brute_force_attack(charset, max_length, target_hash):
    for length in range(1, max_length + 1):
        for guess in itertools.product(charset, repeat=length):
            guess_word = ''.join(guess)
            if hashlib.md5(guess_word.encode()).hexdigest() == target_hash:
                print(f"[+] Password found: {guess_word}")
                return
    print("[-] Password not found")

# Example Usage:
brute_force_attack("abc", 3, "900150983cd24fb0d6963f7d28e17f72")